### PR TITLE
Removed dead switch-cases (Uncreatable Inventories)

### PR DESCRIPTION
**Pull request by thelooter**
# Description
Add Uncreatable Inventories

# Checklist
The following items should be checked before the pull request can be merged.
- [x] Unit tests added.
- [x] Code follows existing code format.

# Info on creating a pull request
- Make sure that unit tests are added which test the relevant changes.
- Make sure that the changes follow the existing code format.

# For maintainer
When a PR is approved, the maintainer should label this PR with `release/*`.

- If the PR fixes a bug in MockBukkit, update the patch version. (if the current version is `0.4.1`, the new version should be `0.4.2`)
- If the PR adds a new feature to MockBukkit, update minor version. (if the current version is `0.4.1`, the new version should be `0.5.0`)

Note that a PR that fixes an `UnimplementedOperationException` should be considered a new version and not a bugfix.

### CODE REVIEW

**Review by Insprill — CHANGES_REQUESTED** [review]
Perhaps InventoryType#isCreatable should be used before the switch?

**Comment by thelooter** [comment]
> Perhaps InventoryType#isCreatable should be used before the switch?

What about filtering the InventoryType Enum to an EnumList that only contains the creatable Inventories and then create the switch from that.

**Comment by Insprill** [comment]
Can you elaborate? I'm not totally sure what you mean since switch cases must be there at compile time

File: src/main/java/be/seeseemelk/mockbukkit/ServerMock.java [hunk 1]
```diff
@@ -642,11 +642,11 @@ public InventoryMock createInventory(InventoryHolder owner, @NotNull InventoryTy
...
 			return new BrewerInventoryMock(owner);
 		case CRAFTING:
-			// TODO: This Inventory Type needs to be implemented
+			throw new UnsupportedOperationException("Crafting Inventory can't be created");
 		case CREATIVE:
-			// TODO: This Inventory Type needs to be implemented
+			throw new UnsupportedOperationException("Creative Inventory can't be created");
 		case MERCHANT:
-			// TODO: This Inventory Type needs to be implemented
+			throw new UnsupportedOperationException("Merchant Inventory can't be created");
```
Comment (Insprill) [inline src/main/java/be/seeseemelk/mockbukkit/ServerMock.java]: There is already an if statement guarding the switch so these cases can never be reached
```suggestion
```

**Review by Insprill — CHANGES_REQUESTED** [review]

**Review by Insprill — APPROVED** [review]
LGTM

**Review by TheBusyBiscuit — APPROVED** [review]

**Pull request merged**